Pr/4

dgillsom opened the pull request.

Commit:
Memory printing functions

Commit:
Second add of printing mem functions

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,61 @@ package sqrl
+	"math/big"
 
-	"github.com/shirou/gopsutil/mem"
+	"github.com/dmmcquay/likexian/host-stat-go"
```
You have this import statement incorrect... It shouldn't be nested inside of my user name.

If you did just a regular:

$go get github.com/likexian/host-stat-go

it would put the package in the right place. 

On another note, just a quick glance at that package, I think we could do something else to get hostname (that is what you are using it for).

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,61 @@ package sqrl
 
-	"github.com/shirou/gopsutil/mem"
+	"github.com/dmmcquay/likexian/host-stat-go"
+	"github.com/dmmcquay/sqrl/dustin/go-humanize"
```
Same thing here, it shouldn't be nested inside of my user name.

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,61 @@ package sqrl
-func MemInfo() {
+func meminfo() (uint64, uint64) {
 	v, _ := mem.VirtualMemory()
+	var total uint64
```
It would be more go-like if you just did:

total := v.Total

Basically, an extra line of code that isn't needed.

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,61 @@ package sqrl
 
-	fmt.Println(v)
+func versioninfo() (string, string, string) {
+	i, _ := host_stat.GetHostInfo()
```
Same thing here.  Don't ignore errors! They can come back to haunt us....

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,61 @@ package sqrl
+	version := i.OSRelease
+	return osVersion, hostname, version
+}
+func mem() {
```
I totally understand what you are trying to do here and for the first shot, it is a great attempt.  Let me give you some ideas of how I want you to move forward.

Lets go ahead and define a struct that encompasses all the data you need to get (well, two actually).  

```
type Ram struct {
    total, free, swapSpace uint64
}
```

```
type OS struct {
    hostname, kernel, version string
} 
```

Create me a function that I can call and it will return me a OS struct and a Ram struct.

Hopefully that makes sense...

dmmcquay commented:
Try fixing some of the comments I made and feel free to ask questions.

Thanks!

Commit:
Added structs and corrected import systems

Commit:
Added a print statement

Commit:
reinitialized files

dgillsom commented:
I added the structs and fixed import statements, but the host-stat package is the only one I can seem to get the hostname information from. The host package of gopsutil errors out every time I call any of the functions from that package, so I had to use the host-stat.

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,57 @@ package sqrl
+func ramOSInfo() (Ram, OS) {
+	v, _ := mem.VirtualMemory()
+	s, _ := mem.SwapMemory()
+	i, _ := host_stat.GetHostInfo()
```
Use https://golang.org/pkg/os/#Hostname for getting the hostname.

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,57 @@ package sqrl
+	swap := mem.swapSpace
+	totalint := int64(total)
+	totalbig := big.NewInt(totalint)
+	totalmb := humanize.BigBytes(totalbig)
```
instead of doing the humanize here, I think it would be better to do that in the functions that create the struct.  The way you have it now, if a user called your function, they would then have to do more work to it.  You printInfo() should just be:

``` go
func printInfo() {
    r, o := ramOSInfo()
    fmt.Println(r)
    fmt.Println(o)
}
```

dmmcquay commented:
Nice work!  Just fix up those last things and I think it'll be ready to go!

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,57 @@ package sqrl
 
-	fmt.Println(v)
+func ramOSInfo() (Ram, OS) {
+	v, _ := mem.VirtualMemory()
```
why aren't you checking the errors here? https://blog.golang.org/error-handling-and-go

Commit:
Checked errors, changed hostname package, moved humanize out of printInfo function

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,60 @@ package sqrl
+func ramOSInfo() (Ram, OS) {
+	v, e := mem.VirtualMemory()
+	if e != nil {
+		fmt.Println(e)
```
This is much better, but still poses an issue. 

So lets say I call this function.  How would the caller of this function know that something happened? Just on stdout, which wouldn't serve them very well inside of the program.  What you should probably do here is something more along the lines of:

``` go
if err != nil {
    return Ram{}, OS{}, err
}
```

and then the caller of this function can know that something went wrong when calling it.  Does that make sense?

dgillsom replied:
I'm not sure what you mean here. Do you mean I should have a third return value for that function? And if the error is nil, then would you still want that returned?

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,60 @@ package sqrl
+}
+
+func ramOSInfo() (Ram, OS) {
+	v, e := mem.VirtualMemory()
```
use the same variable for all the errors.  I typically just use err.

I understand why you did what you did, but with what i commented below, you can get rid of it.

dmmcquay commented on mem.go:
```diff
@@ -2,12 +2,60 @@ package sqrl
+	total := v.Total
+	free := v.Free
+	swap := s.Free
+	totalint := int64(total)
```
Lets make these next three lines more simple.

``` go
totalint := int64(total)
totalbig := big.NewInt(totalint)
totalmb := humanize.BigBytes(totalbig)
```

can just become:

``` go
totalmb := humanize.BigBytes(big.NewInt(int64(total)))
```

Do that for the others ones as well, please.

Commit:
Updated with json, error returns, more swap info
